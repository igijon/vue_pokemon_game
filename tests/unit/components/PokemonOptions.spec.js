import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";

import { pokemons } from "../mocks/pokemons.mock";
//Con yarn test:unit /PokemonOptions sólo hace de ese componente, si no ponemos / ejecuta también
//los test de getPokemonOptions
describe("PokemonOptions component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        //pokemons:pokemon
        pokemons,
      },
    });
  });

  test("debe hacer match con snapshot", () => {
    console.log(wrapper.html());
    expect(wrapper.html()).toMatchInlineSnapshot();
  });
});
