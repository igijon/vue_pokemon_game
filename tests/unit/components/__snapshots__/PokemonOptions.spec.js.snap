// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component debe hacer match con snapshot 1`] = `
<div class="options-container">
  <ul>
    <!--Cuando hagamos click quiero emitir el id del pokemon al padre
      $emit permite comunicar el componente hijo con el padre
      Hay custom events que podemos ver en la documentaciÃ³n de Vue:
      https://vuejs.org/guide/components/events.html
      -->
    <li>pikachu</li>
    <li>charmander</li>
    <li>venusaur</li>
    <li>mew</li>
  </ul>
</div>
`;
